commit/merkleroot/outcome.go

```diff
@@ -46,9 +46,14 @@ func (w *Processor) getOutcome(
 
 	switch nextState {
 	case SelectingRangesForReport:
-		return reportRangesOutcome(commitQuery, consensusObservation), nextState
+		return reportRangesOutcome(q, consensusObservation), nextState
 	case BuildingReport:
-		return buildReport(commitQuery, consensusObservation, previousOutcome), nextState
+		if q.RetryRMNSignatures {
+			// We want to retry getting the RMN signatures on the exact same outcome we had before.
+			// The current observations should all be empty.
+			return previousOutcome, BuildingReport
+		}
```

Comment:
> How many rounds might we stay in the RetryRMNSignatures state?

The way it works right now we retry forever until we get the signatures.
There are some improvements we can do here though.

Reply:
IIRC the "wait for transmission" logic would only retry for 5 rounds. Something like that is needed so that the system would eventually recover from errors.

Reply:
But still, I believe it will end up requesting signatures, just for bigger intervals.

Reply:
What we can do is move forward only with the ones we got enough sigs